### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by blocknotes**
Generic performance improvements after a basic analysis with `rack-mini-profiler` and other gems.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: app/views/spree/components/products/_products.html.erb [hunk 1]
```diff
@@ -9,7 +9,8 @@
   <% end %>
 <% end %>
 
-<% if products.empty? %>
+<% no_products_found = products.empty? %>
```
Comment (kennyadsl) [inline app/views/spree/components/products/_products.html.erb:12]: What about moving this on top of the file where other variables are set?

Comment (blocknotes) [reply app/views/spree/components/products/_products.html.erb:12]: Sure 👍

**Review by kennyadsl — APPROVED** [review]

**Comment by blocknotes** [comment]
@kennyadsl: I think that we could have a flaky spec 😑 with:

_Checkout with attempted XSS escaped XSS string displays the entered state name without evaluating
spec/system/checkout_spec.rb_

**Comment by kennyadsl** [comment]
@blocknotes can you please open an issue about this and link to the CircleCI job failing?

**Pull request merged**